+++ frama-c-Chlorine/src/plugins/wp/wpAnnot.ml	2018-05-31 02:00:00.000000000 +0200
@@ -901,10 +901,15 @@
                 WpStrategy.add_prop_assert b_acc WpStrategy.Ahyp kf s a p
               in (b_acc, (a_acc, e_acc))
           | TBRok | TBRpart ->
-              let id = WpPropId.mk_assert_id config.kf s a in
-              let kind = WpStrategy.Aboth (goal_to_select config id) in
-              let b_acc = WpStrategy.add_prop_assert b_acc kind kf s a p in
-              (b_acc, (a_acc, e_acc))
+            let id = WpPropId.mk_assert_id config.kf s a in
+            let kind =
+                 if Wp_parameters.Assert_check_only.get () then
+                   WpStrategy.Agoal
+                 else
+                   WpStrategy.Aboth (goal_to_select config id)
+            in
+            let b_acc = WpStrategy.add_prop_assert b_acc kind kf s a p in
+            (b_acc, (a_acc, e_acc))
         in acc
     | AAllocation (_b_list, _frees_allocates) ->
         (* [PB] TODO *) acc
+++ frama-c-Chlorine/src/plugins/wp/wp_parameters.ml	2018-05-31 02:00:00.000000000 +0200
@@ -889,6 +889,15 @@
   end)
 let () = on_reset Print.clear
 
+let () = Parameter_customize.set_group wp_po
+let () = Parameter_customize.do_not_save ()
+module Assert_check_only =
+  True (struct
+    let option_name = "-wp-assert-check-only"
+    let help =
+      "Only attempt to prove assertions, do not use them as hypotheses in other proof obligations."
+  end)
+
 (* -------------------------------------------------------------------------- *)
 (* --- OS environment variables                                           --- *)
 (* -------------------------------------------------------------------------- *)
+++ frama-c-Chlorine/src/plugins/wp/wp_parameters.mli	2018-05-31 02:00:00.000000000 +0200
@@ -132,6 +132,8 @@
 module ReportName: Parameter_sig.String
 module Separation: Parameter_sig.Bool
 module Check: Parameter_sig.Bool
+module Assert_check_only: Parameter_sig.Bool
+
 
 (** {2 Environment Variables} *)
 
